ateam_geometry/include/ateam_geometry/variant_do_intersect.hpp

```diff
@@ -43,10 +44,31 @@ namespace ateam_geometry
 template<typename ObjA>
 bool variantDoIntersect(const ObjA & object_a, const AnyShape & object_b)
 {
-  return std::visit(
+  // Use our custom disk intersection if ObjA is a cirlce
+  if constexpr (std::is_same_v<ObjA, Circle>) {
+    if (std::holds_alternative<Circle>(object_b)) {
+      return diskDiskIntersection(object_a, std::get<Circle>(object_b));
+    } else if (std::holds_alternative<Rectangle>(object_b)) {
+      return diskRectangleIntersection(object_a, std::get<Rectangle>(object_b));
+    } else if (std::holds_alternative<Point>(object_b)) {
+      return diskPointIntersection(object_a, std::get<Point>(object_b));
+    }
+  } else {
+    // Also use our custom disk intersection functions if objB
+    // is a circle
+    if (std::holds_alternative<Circle>(object_b)) {
+      if constexpr (std::is_same_v<ObjA, Rectangle>) {
+        return diskRectangleIntersection(std::get<Circle>(object_b), object_a);
+      } else if constexpr (std::is_same_v<ObjA, Point>) {
+        return diskPointIntersection(std::get<Circle>(object_b), object_a);
+      }
+    }
+  }
+  bool intersects = std::visit(
     [&object_a](const auto & b) {
       return CGAL::do_intersect(object_a, b);
     }, object_b);
+  return intersects;
```

Comment:
If we change our disk intersection functions to have the same name (ie. `ateam_geometry::do_disk_intersect()`) then we can simplify this function to just use `std::visit`. That also means we won't have to edit this function every time we add a new disk intersection override.

```suggestion
  if constexpr (std::is_same_v<ObjA, Circle>) {
    return std::visit([&object_a](const auto & b) {
      return ateam_geometry::do_disk_intersect(object_a, b);
    });
  }
  if constexpr (std::is_same_v<ObjB, Circle>) {
    return std::visit([&object_b](const auto & a) {
      return ateam_geometry::do_disk_intersect(object_b, a);
    });
  }
  return std::visit([&object_a](const auto & b) {
    return CGAL::do_intersect(object_a, b);
  });
```

Reply:
Did this but with slightly tweaked formatting since object_a is templated and object_b is a variant